bentoml/artifact/onnx_model_artifact.py

```diff
@@ -0,0 +1,186 @@
+    >>>     @bentoml.api(DataframeHandler)
+    >>>     def predict(self, df):
+    >>>         input_data = df.to_numpy().astype(numpy.float32
+    >>>         input_name = self.artifacts.model.get_inputs()[0].name
+    >>>         output_name = self.artifacts.model.get_outputs()[0].name
+    >>>         return self.artifacts.model.run(
+    >>>                     [output_name], {input_name: input_data}
+    >>>                )[0]
+    >>>
+    >>> svc = OnnxIrisClassifierService()
+    >>>
+    >>> # Option one: pack with path to model on local system
+    >>> svc.pack('model', './rf_iris.onnx')
+    >>>
+    >>> # Option two: pack with ONNX model object
+    >>> # svc.pack('model', onnx_model)
+    >>>
+    >>> # Save BentoService
+    >>> svc.save()
+    """
+
+    def __init__(self, name, backend='onnx'):
+        super(OnnxModelArtifact, self).__init__(name)
+        self.backend = backend
+
+    def _model_file_path(self, base_path):
+        return os.path.join(base_path, self.name + '.onnx')
+
+    def pack(self, obj):  # pylint:disable=arguments-differ
+        try:
+            import onnx
+        except ImportError:
+            return _ExportedOnnxModelArtifact(self, obj)
+
+        if isinstance(obj, onnx.ModelProto):
+            return _OnnxModelArtifactWrapper(self, obj)
+        else:
+            return _ExportedOnnxModelArtifact(self, obj)
+
+    def load(self, path):
+        if self.backend == 'onnx':
+            inference_model = _load_onnx_saved_model(self._model_file_path(path))
+        else:
+            raise NotImplementedError(
+                f'{self.backend} as onnx runtime is not supported at the moment'
+            )
+        return self.pack(inference_model)
+
+    @property
+    def pip_dependencies(self):
+        dependencies = []
+        if self.backend == 'onnx':
+            dependencies.append('onnxruntime')
+        return dependencies
+
+
+class _OnnxModelArtifactWrapper(BentoServiceArtifactWrapper):
+    def __init__(self, spec, model):
+        super(_OnnxModelArtifactWrapper, self).__init__(spec)
+        if not self._model:
+            if self.spec.backend == 'onnxruntime':
+                self._model = _load_onnx_saved_model(model)
+            else:
+                raise NotImplementedError(
+                    f'{self.spec.backend} as onnx runtime is not supported '
+                    f'at the moment'
+                )
+
+    def get(self):
+        return self._model
+
+    def save(self, dst):
+        try:
+            import onnx
+        except ImportError:
+            raise MissingDependencyException(
+                'onnx package is required to save onnx.ModelProto object'
+            )
+        return onnx.save(self._model, self.spec._model_file_path(dst))
+
+
+class _ExportedOnnxModelArtifact(BentoServiceArtifactWrapper):
+    def __init__(self, spec, path):
+        super(_ExportedOnnxModelArtifact, self).__init__(spec)
+
+        self.path = path
```

Comment:
verify the onnx file exist here? a simple isfile and `.onnx` extension check could work

Reply:
model file can end with any extension.

Reply:
might be best to add a `is_onnx_file(file_path)` utility function, and do the verification within the `pack` method above